src/earn/prepareTransactions.ts

```diff
@@ -0,0 +1,117 @@
+import BigNumber from 'bignumber.js'
+import aavePool from 'src/abis/AavePoolV3'
+import erc20 from 'src/abis/IERC20'
+import { TokenBalance } from 'src/tokens/slice'
+import { fetchWithTimeout } from 'src/utils/fetchWithTimeout'
+import { publicClient } from 'src/viem'
+import { TransactionRequest, prepareTransactions } from 'src/viem/prepareTransactions'
+import networkConfig, { networkIdToNetwork } from 'src/web3/networkConfig'
+import { Address, encodeFunctionData, isAddress, parseUnits } from 'viem'
+
+type SimulatedTransactionResponse = {
+  status: 'OK'
+  simulatedTransactions: {
+    status: 'success' | 'failure'
+    blockNumber: string
+    gasNeeded: number
+    gasUsed: number
+    gasPrice: string
+  }[]
+}
+
+export async function prepareSupplyTransactions({
+  amount,
+  token,
+  walletAddress,
+  feeCurrencies,
+  poolContractAddress,
+}: {
+  amount: string
+  token: TokenBalance
+  walletAddress: Address
+  feeCurrencies: TokenBalance[]
+  poolContractAddress: Address
+}) {
+  const baseTransactions: TransactionRequest[] = []
+
+  // amount in smallest unit
+  const amountToSupply = parseUnits(amount, token.decimals)
+
+  if (!token.address || !isAddress(token.address)) {
+    // should never happen
+    throw new Error('Cannot use a token without address')
```

Comment:
Could we add the tokenId to this error to help with debugging if we run into issues?

Reply:
done